### PR TITLE
Fix overriding consensus parameters in evaluator.

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

We failed to override the consensus parameters in COW. This PR fixes it. Background: overriding of consensus parameters is used by Indexer to retrieve the asset close amount of a transaction even when the protocol doesn't support it.

## Test Plan

Add a unit test.

### CODE REVIEW

File: ledger/cow_test.go [hunk 1]
```diff
@@ -112,7 +113,7 @@ func TestCowBalance(t *testing.T) {
 	accts0 := randomAccounts(20, true)
 	ml := mockLedger{balanceMap: accts0}
 
-	c0 := makeRoundCowState(&ml, bookkeeping.BlockHeader{}, 0, 0)
+	c0 := makeRoundCowState(&ml, bookkeeping.BlockHeader{}, config.ConsensusParams{}, 0, 0)
```
Comment (tsachiherman) [inline ledger/cow_test.go:116]: please avoid passing this `config.ConsensusParams{}`, even if that works right now.
Just pick the ConsensusParams for the default protocol version.

File: ledger/eval_test.go [hunk 2]
```diff
@@ -856,7 +856,7 @@ func TestCowCompactCert(t *testing.T) {
 	blocks := make(map[basics.Round]bookkeeping.BlockHeader)
 	blockErr := make(map[basics.Round]error)
 	ml := mockLedger{balanceMap: accts0, blocks: blocks, blockErr: blockErr}
-	c0 := makeRoundCowState(&ml, bookkeeping.BlockHeader{}, 0, 0)
+	c0 := makeRoundCowState(&ml, bookkeeping.BlockHeader{}, config.ConsensusParams{}, 0, 0)
```
Comment (tsachiherman) [inline ledger/eval_test.go:859]: same as above.

File: ledger/cow_test.go (hunk 1 above)
Comment (tolikzinovyev) [reply ledger/cow_test.go]: done

File: ledger/eval_test.go (hunk 2 above)
Comment (tolikzinovyev) [reply ledger/eval_test.go]: done

**Review by tsachiherman — APPROVED** [review]
Looks good to me.

**Pull request merged**